megamek/src/megamek/common/Compute.java

```diff
@@ -4290,84 +4272,101 @@ private static int calcSensorShadow(IGame game, Entity ae,
     }
 
     /**
-     * Checks to see if an entity has passed out of range of a previously established firing solution
+     * Updates an entity's firingSolutions, removing any objects that no longer meet criteria for being
+     * tracked as targets. Also, if the detecting entity no longer meets criteria for having firing solutions,
+     * empty the list. We wouldn't want a dead ship to be providing NC3 data, now would we...
      */
-    public static void removeFiringSolution(Entity detector) {
-        Vector<Entity> toRemove = new Vector<Entity>();
-        //If the detector is dead, has no position, or has flown offboard, remove everything
+    public static void updateFiringSolutions(IGame game, Entity detector) {
+        List<Integer> toRemove = new ArrayList<Integer>();
+        //Flush the detecting unit's firing solutions if any of these conditions applies
         if (detector.isDestroyed()
+                || detector.isDoomed()
+                || detector.getTransportId() != Entity.NONE
+                || detector.isPartOfFighterSquadron()
                 || detector.isOffBoard()
                 || detector.getPosition() == null) {
-            detector.firingSolutions.removeAllElements();
+            detector.clearFiringSolutions();
             return;
         }
-        for (Entity target : detector.firingSolutions) {
-            //If the target is dead, has no position or has flown offboard, remove it
+        for (int id : detector.getFiringSolutions()) {
+            Entity target = game.getEntity(id);
+            //The target should be removed if it's off the board for any of these reasons
```

Comment:
Like above, will target always be non-null?

Reply:
Good catch. I had added a null guard to a couple of similar checks, but missed this one.